test/KnoBlockIO.ts

```diff
@@ -1,260 +1,260 @@
-        await instance.create(1001, one);
-        await instance.deposit(zero, 1000);
-        expect(await instance.currentAmount(zero)).to.equal(
-          BigNumber.from('1000'),
-        );
-      });
-      it('updates msg.senders deposit value accurately', async function () {
-        await instance.create(1001, one);
-        await instance.deposit(zero, 1000);
-        expect(await instance.deposits(zero, deployer.address)).to.equal(
-          BigNumber.from('1000'),
-        );
-      });
-      it('emits unlock event if currentAmount is equal to UnlockAmount', async function () {
-        await instance.create(1001, one);
-        expect(await instance.deposit(zero, 1001))
-          .to.emit(instance, 'BlockUnlocked')
-          .withArgs(zero);
-      });
-      it('emits unlock event if currentAmount over to UnlockAmount', async function () {
-        const msgvalue = {
-          value: ethers.utils.parseEther('0.000000000000002000'),
-        }; //sending 2000 wei
-        await instance.create(1001, one);
-        expect(await instance.deposit(zero, 2000, msgvalue))
-          .to.emit(instance, 'BlockUnlocked')
-          .withArgs(zero);
-      });
+        it('sets creator as msg.sender', async function () {
+          await instance.create(1001, one);
+          expect(await instance.creator(zero)).to.equal(deployer.address);
+        });
 
-      it('returns deposit overkill to Msg.sender', async function () {
-        const msgvalue = {
-          value: ethers.utils.parseEther('0.000000000000002000'),
-        }; //sending 2000 wei
-        await instance.create(1001, one);
-        expect(
-          await instance.deposit(zero, 2000, msgvalue),
-        ).to.changeEtherBalance(deployer, -msgvalue);
+        it('sets unlockAmount to correctly', async function () {
+          await instance.create(1001, one);
+          expect(await instance.unlockAmount(zero)).to.equal(
+            BigNumber.from('1001'),
+          );
+        });
+
+        it('sets knoType to correctly', async function () {
+          await instance.create(1001, one);
+          expect(await instance.knoType(zero)).to.equal(one);
+        });
+        it('emits new KnoBlock Event', async function () {
+          await expect(instance.create(1001, one))
+            .to.emit(instance, 'KnoBlockCreated')
+            .withArgs(zero);
+        });
       });
+      describe('#deposit()', () => {
+        it('increases the KnoBlocks currentAmount by msg.value - fee', async function () {
+          const msgvalue = ethers.utils.parseUnits('1001', 0);
+          await instance.create(1001, one);
+          await instance.setDepositFee(185);
+          await instance.deposit(zero, { value: msgvalue });
 
-      describe('reverts if...', () => {
-        it('fails transaction in KnoBlock is already Unlocked', async function () {
+          const depositedAmount = msgvalue;
+          const fee = depositedAmount.mul(185).div(10000);
```

Comment:
You should read the `feeBP` from the contract and use it - avoid hardcoded values and magic numbers.

Reply:
Replace `10000` with a constant at the top called `BASIS`